### PR TITLE
layers: Label VUID 04145

### PR DESCRIPTION
**Pull request by sfricke-samsung**
VUID-VkPipelineShaderStageCreateInfo-module-04145

### CODE REVIEW

**Comment by mark-lunarg** [comment]
Running through CI.

**Comment by mark-lunarg** [comment]
@sfricke-samsung, this failed on 32- and 64-bit NV Windows platforms with the following relevant log output:

```
[ RUN      ] VkLayerTest.CooperativeMatrixNV
Unexpected: Validation Error: [ VUID-VkPipelineShaderStageCreateInfo-module-04145 ] Object 0: handle = 0x1d37f6fdd88, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0xab738885 | After specialization was applied, VkShaderModule 0x3ba5830000000006[] does not contain valid spirv for stage VK_SHADER_STAGE_COMPUTE_BIT. The Vulkan spec states: The SPIR-V code that was used to create module must be valid as described by the Khronos SPIR-V Specification after applying the specializations provided in pSpecializationInfo, if any, and then converting all specialization constants into fixed constants. (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkPipelineShaderStageCreateInfo-module-04145)
Unexpected: Validation Error: [ UNASSIGNED-CoreValidation-Shader-CooperativeMatrixType ] Object 0: VK_NULL_HANDLE, type = VK_OBJECT_TYPE_PIPELINE; | MessageID = 0x658db585 | OpTypeCooperativeMatrixNV (result id = 10) operands don't match a supported matrix type
Unexpected: Validation Error: [ UNASSIGNED-CoreValidation-Shader-CooperativeMatrixMulAdd ] Object 0: VK_NULL_HANDLE, type = VK_OBJECT_TYPE_PIPELINE; | MessageID = 0xf9bd0ee3 | OpCooperativeMatrixMulAddNV (result id = 25) operands don't match a supported matrix VkCooperativeMatrixPropertiesNV
Other error messages logged for this test were:
     Validation Error: [ VUID-VkPipelineShaderStageCreateInfo-module-04145 ] Object 0: handle = 0x1d37f6fdd88, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0xab738885 | After specialization was applied, VkShaderModule 0x3ba5830000000006[] does not contain valid spirv for stage VK_SHADER_STAGE_COMPUTE_BIT. The Vulkan spec states: The SPIR-V code that was used to create module must be valid as described by the Khronos SPIR-V Specification after applying the specializations provided in pSpecializationInfo, if any, and then converting all specialization constants into fixed constants. (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkPipelineShaderStageCreateInfo-module-04145)
     Validation Error: [ UNASSIGNED-CoreValidation-Shader-CooperativeMatrixType ] Object 0: VK_NULL_HANDLE, type = VK_OBJECT_TYPE_PIPELINE; | MessageID = 0x658db585 | OpTypeCooperativeMatrixNV (result id = 10) operands don't match a supported matrix type
     Validation Error: [ UNASSIGNED-CoreValidation-Shader-CooperativeMatrixMulAdd ] Object 0: VK_NULL_HANDLE, type = VK_OBJECT_TYPE_PIPELINE; | MessageID = 0xf9bd0ee3 | OpCooperativeMatrixMulAddNV (result id = 25) operands don't match a supported matrix VkCooperativeMatrixPropertiesNV
C:\j\w8\64-bit\Debug\tests\vkrenderframework.cpp(185): error: Failed
Did not receive expected error 'VUID-VkPipelineShaderStageCreateInfo-module-parameter'
[  FAILED  ] VkLayerTest.CooperativeMatrixNV (1626 ms)
```

File: tests/vklayertests_pipeline_shader.cpp [hunk 1]
```diff
@@ -5887,7 +5887,7 @@ TEST_F(VkLayerTest, CooperativeMatrixNV) {
     m_errorMonitor->VerifyFound();
     m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "UNASSIGNED-CoreValidation-Shader-CooperativeMatrixType");
     m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "UNASSIGNED-CoreValidation-Shader-CooperativeMatrixMulAdd");
-    m_errorMonitor->SetUnexpectedError("VUID-VkPipelineShaderStageCreateInfo-module-parameter");
```
Comment (sfricke-samsung) [inline tests/vklayertests_pipeline_shader.cpp:5890]: @mark-lunarg So 5 lines above it has

```
m_errorMonitor->SetDesiredFailureMsg(kErrorBit, "VUID-VkPipelineShaderStageCreateInfo-module-parameter");
```

and I made the guess (as have no local way to test this test as it skips devsim) and I think the `SetDesiredFailureMsg` needs to be `04145`

Reverting this line and applying to the one above will probably fix this, will push patch now

**Comment by mark-lunarg** [comment]
Running through the farm again...

**Comment by mark-lunarg** [comment]
@sfricke-samsung, still getting a Windows NV failure as follows.  My hunch is that the error message that was in the `CooperativeMatrixNV `test is still being generated in a different check (it shows up in a couple more places I think), or something like that.

```
[ RUN      ] VkLayerTest.CooperativeMatrixNV
Unexpected: Validation Error: [ VUID-VkPipelineShaderStageCreateInfo-module-04145 ] Object 0: handle = 0x1ad4d8915d8, type = VK_OBJECT_TYPE_DEVICE; | MessageID = 0xab738885 | After specialization was applied, VkShaderModule 0x3ba5830000000006[] does not contain valid spirv for stage VK_SHADER_STAGE_COMPUTE_BIT. The Vulkan spec states: The SPIR-V code that was used to create module must be valid as described by the Khronos SPIR-V Specification after applying the specializations provided in pSpecializationInfo, if any, and then converting all specialization constants into fixed constants. (https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#VUID-VkPipelineShaderStageCreateInfo-module-04145)
C:\j\w8\64-bit\Debug\tests\vkrenderframework.cpp(195): error: Failed
Received unexpected error(s).
[  FAILED  ] VkLayerTest.CooperativeMatrixNV (1188 ms)
```

**Comment by sfricke-samsung** [comment]
@mark-lunarg I have no way to actually check this (lack of window machine with Nvidia card with extension supported) and I don't know if it's worth me to keep guessing and having you to keep checking or if this is possible for someone with access to a valid machine can look into

**Comment by mark-lunarg** [comment]
@sfricke-samsung, if it was just about anybody else the answer would be 'no', but I'll take a look tomorrow...

**Comment by mark-lunarg** [comment]
Turned out to be a simple change, but a pain to find but it's passing tests and is in now -- thanks for your patience.  Also, you might hint to the greater Samsung powers that a high-end Windows machine and a couple of souped-up graphics cards are critical for your workflow!

**Pull request closed without merging**